app/src/main/java/com/example/weather/data/RepositoryImpl.java

```diff
@@ -109,8 +109,29 @@ public Completable updateWeather(WeatherData weatherData) {
     }
 
     @Override
-    public SettingManager getSettingsMenager() {
-        return this.settingManager;
+    public Observable<String> getTimezone(String lat, String lon) {
+        return ForecastApi.Instance.getForecastApi().getWeatherDataByCoord(lat,lon)
+                .map(WeatherEntity::getTimezone);
+    }
+
+
+    @Override
+    public void loadLocale(Context context) {
+        if(settingManager==null) this.settingManager=new SettingManager(context);
```

Comment:
Why do you do that? As I see, Repository is Singleton. Moreover, when you init Repository you pass Context, so you don't need to pass Context here. You can just init SettingManager with Repository.

Reply:
To change the language, I need the context from the main activity. The repository is initialized in the app.